### PR TITLE
Travis CI integration

### LINKED ISSUE #24: [Feature Request] Use travis ci config to publish docker image
Currently docker images are being pushed by hand. This ticket is to use travis ci config to publish the docker image.
We need to think about how we go about promoting releases too.

### PR DESCRIPTION
**Pull request by 0001vrn**
### :pencil: Description
- Added Travis CI config to push docker images for the master branch
- Encrypted docker credentials for secure communication
- Closes #24 

**Testing notes:**
- Successfully tested with my personal Travis CI account and docker hub account
https://travis-ci.com/github/0001vrn/pitchfork/builds/175271353
https://hub.docker.com/repository/docker/vthakurdocker/pitchfork/tags?page=1

### :link: Related Issues
- Need to add docker credentials (DOCKER_PASSWORD, DOCKER_USERNAME)  with official Travis ci account
- Need to encrypt the docker credentials
- Need appropriate email Id for Travis CI notifications

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+before_install:
+- echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+
+script:
+- if ([ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]) || [ -n "$TRAVIS_TAG" ]; then make release; else make test; fi
+
+notifications:
+  email:
+  - vthakur@expedia.com
```
Comment (sumitanvekar) [inline .travis.yml:28]: these values might have to be changed.

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
 test:
-	./mvnw clean verify
+	${MAVEN} clean verify
+
+package:
+	${MAVEN} clean package
 
 build:
-	docker build -t expediagroup/pitchfork:test . -f Dockerfile
+	docker build -t vthakurdocker/pitchfork:latest -t vthakurdocker/pitchfork:1.24 . -f Dockerfile
```
Comment (sumitanvekar) [inline Makefile:12]: these values might have to be changed.

**Comment by worldtiki** [comment]
Hi @0001vrn :)

Thanks for the pr. We recently switched from travis to gh actions and I forgot to update this issue :( A million apologies.

Do you want to take a stab with github actions?

**Comment by 0001vrn** [comment]
Hi @worldtiki 
You're welcome :) 

I think @nikos912000 also mentioned the same around moving to gh actions. Could you update the issue to be more appropriate? and mention the necessary things? I will give it a try.

I will close this PR in favor of GH action one. I hope that should be fine :)

File: Makefile (hunk 2 above)
Comment (0001vrn) [reply Makefile:12]: Eventually, yes

File: .travis.yml (hunk 1 above)
Comment (0001vrn) [reply .travis.yml:28]: Eventually, yes :)

**Pull request closed without merging**